frontend/src/features/properties/map/MapRouter.tsx

```diff
@@ -0,0 +1,117 @@
+import Claims from 'constants/claims';
+import MotiInventoryContainer from 'features/mapSideBar/MotiInventoryContainer';
+import { IPropertyApiModel } from 'interfaces/IPropertyApiModel';
+import queryString from 'query-string';
+import * as React from 'react';
+import { matchPath, Switch, useHistory, useLocation } from 'react-router-dom';
+import AppRoute from 'utils/AppRoute';
+
+import AcquisitionContainer from './acquisition/AcquisitionContainer';
+import AddAcquisitionContainer from './acquisition/add/AddAcquisitionContainer';
+import AddResearchContainer from './research/add/AddResearchContainer';
+import ResearchContainer from './research/ResearchContainer';
+
+interface IMapRouterProps {
+  showSideBar: boolean;
+  setShowSideBar: (show: boolean) => void;
+  onZoom?: (apiProperty?: IPropertyApiModel) => void;
+}
+
+export const MapRouter: React.FunctionComponent<IMapRouterProps> = React.memo(props => {
+  const location = useLocation();
+  const history = useHistory();
+
+  let matched = matchPath(location.pathname, {
+    path: '/mapview/sidebar/*',
+    exact: true,
+    strict: true,
+  });
+
+  React.useEffect(() => {
+    if (matched !== null) {
+      props.setShowSideBar(true);
+    } else {
+      props.setShowSideBar(false);
+    }
+  }, [matched, props]);
+
+  const onClose = () => {
+    history.push('/mapview');
+  };
+
+  const pidQueryString = queryString.parse(location.search).pid?.toString() ?? '';
+  return (
+    <Switch>
+      <AppRoute
+        path={`/mapview/sidebar/research/new`}
+        customRender={() => <AddResearchContainer onClose={onClose} />}
+        claim={Claims.RESEARCH_ADD}
+        exact
+        key={'NewResearch'}
+        title={'Create Research File'}
+      />
+      <AppRoute
+        path={`/mapview/sidebar/research/:researchId`}
+        customRender={({ match }) => (
```

Comment:
ok, looking at this now I think previously we would have handled this with a <Page><Container /></Page> strictly speaking I don't think I care either way on this one, it did always seem like <Page> components never had much to do.

Reply:
I thought about that too, but given this is not really a page and more like a widget I decided to not go that route. But it would be totally valid for a page